File: SYP.Web/Modules/Setting/VatRates/VatRatesColumns.cs
namespace SYP.Setting.Columns;

[ColumnsScript("Setting.VatRates")]
[BasedOnRow(typeof(VatRatesRow), CheckNames = true)]
public class VatRatesColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Code { get; set; }
    public bool IsDefault { get; set; }
    public bool IsActive { get; set; }
    public int SortOrder { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Setting/VatRatesColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { VatRatesRow } from "./VatRatesRow";

export interface VatRatesColumns {
    Id: Column<VatRatesRow>;
    Name: Column<VatRatesRow>;
    Code: Column<VatRatesRow>;
    IsDefault: Column<VatRatesRow>;
    IsActive: Column<VatRatesRow>;
    SortOrder: Column<VatRatesRow>;
}

export class VatRatesColumns extends ColumnsBase<VatRatesRow> {
    static readonly columnsKey = 'Setting.VatRates';
    static readonly Fields = fieldsProxy<VatRatesColumns>();
}